### PR TITLE
advanced logging configuration

### PR DESCRIPTION
**Pull request by ekneg54**
this PR aims to add more flexibility to the configuration of logpreps logging facility.
it implements the python dictConfig approach described here in the [python logging documentation](https://docs.python.org/3/library/logging.config.html#logging.config.dictConfig).

As we already use custom names for our loggers I decided to leave this contrary the best practice advice to use `__name__` as the logger name. As a side effect we do not have a hierarchical order of our loggers anymore, but it gives us the flexibility to independently configure the log level of every logger in the configuration now. But I shortened the logger names in favor to get a better log output.

GOALS:

- [x] configurable logging format
- [x] add hostname to log entries
- [x] configurable date format
- [x] set level per log instance (root logger is DEBUG, but Pipeline 1 logs only ERROR)

TODO:

- [x] add tests
- [x] add documentation
- [x] add changelog
- [x] get rid of duplicate log messages

### CODE REVIEW

File: tests/unit/util/test_logging.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+from logprep.util.defaults import DEFAULT_LOG_CONFIG
+from logprep.util.logging import LogprepFormatter
+
+
+class TestLogprepFormatter:
+
+    def test_default_log_config(self):
+        logging.config.dictConfig(DEFAULT_LOG_CONFIG)
+        logger = logging.getLogger("test")
+        assert logger
```
Comment (dtrai2) [inline tests/unit/util/test_logging.py:16]: doesn't this assertion only check if the logger object exists, but not if the default configuration was actually applied?
```suggestion
        assert isinstance(logger.root.handlers[0].formatter, LogprepFormatter)
```

File: tests/unit/util/test_logging.py [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+    def test_formatter_init_with_default(self):
+        default_formatter_config = DEFAULT_LOG_CONFIG["formatters"]["logprep"]
+        formatter = LogprepFormatter(
+            fmt=default_formatter_config["format"], datefmt=default_formatter_config["datefmt"]
+        )
+        assert formatter
```
Comment (dtrai2) [inline tests/unit/util/test_logging.py:23]: similar as the previous comment

**Review by dtrai2 — CHANGES_REQUESTED** [review]
I had a look at it and only found some minor suggestions. The first thing I noticed is though, that when running logprep with `python logprep/run_logprep.py run ./quickstart/exampledata/config/pipeline.yml` I get a stacktrace telling me that the `FileNotFoundError ... multiprocdir ...`. Logprep should rather tell me in a propper logmessage that the multiprocdir is missing, at least the stacktrace shouldn't be visible.

File: tests/testdata/config/config.yml [hunk 3]
```diff
@@ -44,9 +44,9 @@ input:
     kafka_config:
       bootstrap.servers: "127.0.0.1:9092"
       group.id: "cgroup"
-      auto.commit: "true"
-      session.timeout: "6000"
-      offset.reset.policy: "smallest"
+      enable.auto.commit: "true"
+      session.timeout.ms: "6000"
+      auto.offset.reset: "smallest"
```
[the remark this answers is not in the document]
Comment (ekneg54) [reply tests/testdata/config/config.yml:49]: the new kafka image does not support these old config parameters anymore

File: logprep/runner.py [hunk 4]
```diff
@@ -147,15 +147,18 @@ def start(self):
         self._manager.restart()
         self._logger.info("Startup complete")
         self._logger.debug("Runner iterating")
+        self._iterate()
+        self._logger.info("Shutting down")
+        self._logger.info("Initiated shutdown")
```
[the remark this answers is not in the document]
Comment (ekneg54) [reply logprep/runner.py]: done

**Review by dtrai2 — APPROVED** [review]

File: tests/testdata/config/config.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (dtrai2) [reply tests/testdata/config/config.yml:49]: ah okay, thanks for the explanation.

**Pull request merged**